app/controllers/api/v1/ability.rb

```diff
@@ -96,7 +97,21 @@ def beneficiary_abilities
       end
 
       def canned_response_abilities
-        can %I[index create update destroy show], CannedResponse if can_manage_canned_response?
+        can %i[index create update destroy show], CannedResponse if can_manage_canned_response?
+      end
+
+      def offer_response_abilities
+        can :create, OfferResponse do |offer_response|
+          offer_response.user_id == @user_id &&
+            Shareable.non_expired.find_by(
+              resource_type: 'Offer',
+              resource_id: offer_response.offer_id
+            ).present?
+        end
+
+        can %i[index show], OfferResponse
```

Comment:
```suggestion
        can %i[index show], OfferResponse, { user_id: @user_id }
```
A user should only be allowed to view his/her records

Reply:
Note: make sure there's a test for it : )